giotto/diagrams/features.py

```diff
@@ -516,3 +516,122 @@ def transform(self, X, y=None):
+    [b, d, q], subdiagrams corresponding to distinct homology dimensions are
+    considered separately. For each subdiagram, the number of relevant holes is
+    found. A hole is relevant if its lifetime d - b is larger than a given
+    threshold.
+
+    Parameters
+    ----------
+    threshold_type : ``'rel'`` | ``'abs'``, optional, default: ``rel``
+        - If ``threshold_type == 'rel'`` the threshold for relevance is
+          relative to the maximum lifetime within a persistence diagram and
+          dimension.
+        - If ``threshold_type == 'abs'`` the parameter ``frac`` defines an
+          absolute threshold.
+
+    threshold_fraction : float, optional, default: ``0.``
+        The fraction of the maximum lifetime or threshold (depends on
+        parameter ``threshold_type``) above which holes are considered to be
+        relevant.
+
+    n_jobs : int or None, optional, default: ``None``
+        The number of jobs to use for the computation. ``None`` means 1 unless
+        in a :obj:`joblib.parallel_backend` context. ``-1`` means using all
+        processors.
+
+    Attributes
+    ----------
+    homology_dimensions_ : list
+        Homology dimensions seen in :meth:`fit`, sorted in ascending order.
+    """
+
+    def __init__(self, threshold_type='rel', threshold_fraction=0.,
+                 n_jobs=None):
+        self.threshold_type = threshold_type
+        self.threshold_fraction = threshold_fraction
+        self.n_jobs = n_jobs
+
+    def _get_n_holes(self, X):
+        # the lifespan of point is given by the difference between its
+        # disappearance and appearance
+        X_lifespan = X[:, :, 1] - X[:, :, 0]
+        if self.threshold_type == 'rel':
+            # return the number of points having a lifespan longer than
+            # a fraction of the maximum lifetime of all holes
+            return np.sum(X_lifespan > self.threshold_fraction *
+                          X_lifespan.max(axis=1)
+                          .reshape(-1, 1),
+                          axis=1)
+        elif self.threshold_type == 'abs':
+            return np.sum(X_lifespan >= self.threshold_fraction, axis=1)
+
+    def fit(self, X, y=None):
+        """
+        Parameters
+        ----------
+        X : ndarray, shape (n_sample, n_features, 3)
+            Input data. Array of persistence diagrams, each a collection of
+            triples [b, d, q] representing persistent topological features
+            through their birth (b), death (d) and homology dimension (q).
+
+        y : None
+            There is no need for a target in a transformer, yet the pipeline
+            API requires this parameter.
+
+        Returns
+        -------
+        self : object
+        """
+        X = check_diagram(X)
+
+        self.homology_dimensions_ = sorted(set(X[0, :, 2]))
+        self._n_dimensions = len(self.homology_dimensions_)
+
+        self._is_fitted = True
+
+        return self
+
+    def transform(self, X):
```

Comment:
I think it would be useful to use the check_array() function from sklearn. It basically checks if the input array is 2-dimensional and if only contains finite values (other parameters can be set).

Reply:
@nphilou , @ulupo : What do you think? This would also have to be changed in other tests that exist in giotto-learn at the moment.

Reply:
Unless I misunderstood @alexbacce's comment, I think the reason we have a custom check_diagram validation function is that our input data structures are not fully compliant with sklearn's -- the main difference being that our diagram arrays tend to be 3-dimensional instead of 2, and have special contents along axis 2 (the homology dimensions). So I'd argue that it is desirable to keep adopting a custom check_diagram function which can be modelled around sklearn's check_array (as the current one was) while implementing specific checks which are important to our data structures.  Would be good to have @gtauzin's opinion on this too.

Reply:
+1 with @ulupo 

I also think that `check_diagram` should contain sklearn's `check_array` to cover more validation cases.

Reply:
Yeah, I see your point @ulupo. I said that because I saw that some of the checks done in  `check_diagram` are already done in `check_array` of sklearn. Therefore, maybe as suggested by @nphilou it would be useful to include `check_array` directly in `check_diagram` (maybe removing the check that enforces the array to be 2-dimensional).

Reply:
@alexbacce and @nphilou: I'm sold on your suggestions. Maybe (by which I mean definitely :P) create an issue on this?